dbms/src/Flash/Mpp/MPPTask.cpp

```diff
@@ -461,24 +480,31 @@ void MPPTask::runImpl()
 void MPPTask::handleError(const String & error_msg)
 {
     auto * manager_ptr = manager.load();
-    /// if manager_ptr is not nullptr, it means the task has already been registered,
-    /// MPPTaskManager::cancelMPPQuery will handle it properly if the query is to be cancelled.
-    if (manager_ptr == nullptr || !manager_ptr->isQueryToBeCancelled(id.start_ts))
+    if (manager_ptr != nullptr)
+    {
+        auto updated_msg = fmt::format("From {}: {}", id.toString(), error_msg);
+        /// task already been registered in TaskManager, use task manager to abort the whole query in this node
+        newThreadManager()->scheduleThenDetach(true, "MPPTaskManagerAbortQuery", [manager_ptr, updated_msg, query_id = id.start_ts] {
```

Comment:
Because `MPPTaskManager::abortMPPQuery` is blocked until all mpp tasks' cancel is finished, which sometimes need to wait until some mpp tasks' destruct is finished. If don't use a new thread, there might be a risk of deadlock since the destruct of current mpp task is not started untill some mpp tasks' destruct is finished.